### PR TITLE
[main][feat]: script for supporting private vault

### PR DESCRIPTION
**Pull request by abc-alpaca**
## Description
- for supporting private vault

### CODE REVIEW

File: deploy/exec/delta-neutral-vault-config/config/set-params.ts [hunk 1]
```diff
@@ -136,15 +135,15 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+      const vaultConfig = DeltaNeutralVaultConfig02__factory.connect(info.VAULT_CONFIG, deployer);
       const setParamsTx = await vaultConfig.setParams(
         info.WRAPPED_NATIVE_ADDRESS,
         info.WNATIVE_RELAYER_ADDRESS,
         info.FAIRLAUNCH_ADDRESS,
         info.REBALANCE_FACTOR,
         info.POSITION_VALUE_TOLERANCE,
         info.DEBT_RATIO_TOLERANCE,
-        { nonce: nonce++ }
+        { nonce: nonce++, gasLimit: 200000000 }
```
Comment (abc-alpaca) [inline deploy/exec/delta-neutral-vault-config/config/set-params.ts:146]:
```suggestion
        { nonce: nonce++, gasLimit: 200000000 }
```
remove gasLimit

File: deploy/exec/av-controller/config/set-param.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+  ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+  ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+  ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+  ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+  Check all variables below before execute the deployment script
+  */
+
+  const CREDITOR_ADDRS: string[] = ["0xBCbA5F83f450a9e358964B4e306339d1F64B630e"];
+  const PRIVATE_VAULT_ADDRS: string[] = ["0xC57876a95A4f31a0A4FDB0329Fc78e00B092cC94"];
```
Comment (abc-alpaca) [inline deploy/exec/av-controller/config/set-param.ts:21]: improve
```suggestion
  const CREDITOR_ADDRS: string[] = ["0xBCbA5F83f450a9e358964B4e306339d1F64B630e"];
  const PRIVATE_VAULT_ADDRS: string[] = ["0xC57876a95A4f31a0A4FDB0329Fc78e00B092cC94"];
```
improve using creditor name and private-vault-symbol

File: deploy/exec/av-controller/config/set-param.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  const deployer = await getDeployer();
+  const config = getConfig();
+
+  // VALIDATING CREDITORS
+  validateInput(CREDITOR_ADDRS, PRIVATE_VAULT_ADDRS);
```
Comment (abc-alpaca) [inline deploy/exec/av-controller/config/set-param.ts:27]:
```suggestion
  validateInput(CREDITOR_ADDRS, PRIVATE_VAULT_ADDRS,config);
```

File: deploy/exec/av-controller/config/set-param.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  if (PRIVATE_VAULT_ADDRS.length > 0) {
+    console.log(`>> Set PRIVATEVAULT TO : ${PRIVATE_VAULT_ADDRS}`);
+    await avController.setPrivateVaults(PRIVATE_VAULT_ADDRS, ops);
+  }
+  const cfh = new ConfigFileHelper();
+  cfh.setCreditToAVController(CREDITOR_ADDRS);
+};
+
+function validateInput(creditors: string[], privateVaults: string[]) {
```
Comment (abc-alpaca) [inline deploy/exec/av-controller/config/set-param.ts:49]:
```suggestion
function validateInput(creditors: string[], privateVaults: string[],config: Config) {
```

File: deploy/exec/creditor/deploy/xalpaca-creditor.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const xalpacaCreditor = (await upgrades.deployProxy(XALPACACreditor, [
+    XALPACA_ADDRESS,
+    VALUE_PER_XALPACA,
+  ])) as XALPACACreditor;
+  console.log(`>> Deployed at ${xalpacaCreditor.address}`);
+
+  const configFile = new ConfigFileHelper();
+
+  configFile.addOrSetCreditors({
+    name: "xAlpacaCreditors",
```
Comment (abc-alpaca) [inline deploy/exec/creditor/deploy/xalpaca-creditor.ts:41]:
```suggestion
    name: "xAlpacaCreditor",
```

File: deploy/exec/delta-neutral-vault-config/config/set-params.ts [hunk 6]
```diff
@@ -66,8 +65,8 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     NEW_PARAMS.map(async (n) => {
       const automatedVault = config.DeltaNeutralVaults.find((v) => v.symbol === n.VAULT_SYMBOL);
       if (automatedVault === undefined) throw new Error(`error: unable to map ${n.VAULT_SYMBOL} to any vault`);
-
-      const vaultConfig = DeltaNeutralVaultConfig__factory.connect(automatedVault.config, deployer);
+      console.log("automatedVault.config", automatedVault.config);
```
Comment (abc-alpaca) [inline deploy/exec/delta-neutral-vault-config/config/set-params.ts:68]:
```suggestion
      console.log("automatedVault.config", automatedVault.config);
```
remove

File: deploy/exec/delta-neutral-vault-config/config/set-controller.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  ░██╗░░░░░░░██╗░█████╗░██████╗░███╗░░██╗██╗███╗░░██╗░██████╗░
+  ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+  ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+  ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+  ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+  ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+  Check all variables below before execute the deployment script
+  */
+
+  const DLTA_VAULT_SYMBOL = ["n8x-BNBUSDT-PCS2"];
```
Comment (abc-alpaca) [inline deploy/exec/delta-neutral-vault-config/config/set-controller.ts:19]:
```suggestion
  const DELTA_VAULT_SYMBOL = ["n8x-BNBUSDT-PCS2"];
```

File: deploy/exec/delta-neutral-vault-config/config/set-controller.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+  const deployer = await getDeployer();
+  const config = getConfig();
+  const CONTROLLER = config.AutomatedVaultController!.address;
+
+  // VALIDATING ALL DLTA_VAULT_SYMBOL
+  const configs = DLTA_VAULT_SYMBOL.map((inputVaultSymbol) => {
```
Comment (abc-alpaca) [inline deploy/exec/delta-neutral-vault-config/config/set-controller.ts:26]:
```suggestion
  // VALIDATING ALL DELTA_VAULT_SYMBOL
  const configs = DELTA_VAULT_SYMBOL.map((inputVaultSymbol) => {
```

File: deploy/exec/delta-neutral-vault-config/config/set-controller.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    const deltaVaultConfig = config.DeltaNeutralVaults.find((o) => compare(o.symbol, inputVaultSymbol));
+    if (!deltaVaultConfig) {
+      throw new Error(`ERROR : DLTA_VAULT_SYMBOL is INVALID : ${inputVaultSymbol}`);
```
Comment (abc-alpaca) [inline deploy/exec/delta-neutral-vault-config/config/set-controller.ts:29]:
```suggestion
      throw new Error(`ERROR : DELTA_VAULT_SYMBOL is INVALID : ${inputVaultSymbol}`);
```

File: deploy/exec/delta-neutral-vault-config/config/set-params.ts [hunk 10]
```diff
@@ -4,10 +4,11 @@ import { ethers } from "hardhat";
 import { TimelockEntity } from "../../../entities";
 import { fileService, TimelockService } from "../../../services";
```
Comment (abc-alpaca) [inline deploy/exec/delta-neutral-vault-config/config/set-params.ts:5]: revert this change

File: utils/deployer-helper.ts [hunk 11]
```diff
@@ -7,13 +7,23 @@ import { HttpNetworkUserConfig } from "hardhat/types";
 export async function getDeployer(): Promise<SignerWithAddress> {
   const [defaultDeployer] = await ethers.getSigners();
 
-  if (network.name === "mainnetfork" || network.name === "fantom_mainnetfork") {
+  if (isFork()) {
     const provider = ethers.getDefaultProvider((network.config as HttpNetworkUserConfig).url) as JsonRpcProvider;
-    const signer = provider.getSigner(defaultDeployer.address);
+    const signer = provider.getSigner(process.env.DEPLOYER_ADDRESS);
```
Comment (abc-alpaca) [inline utils/deployer-helper.ts:12]: add env.example key

File: deploy/exec/av-controller/config/set-param.ts [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  if (pvAddrs.length > 0) {
+    console.log(`>> Set PRIVATEVAULT TO : ${pvAddrs}`);
+    await avController.setPrivateVaults(pvAddrs, ops);
+  }
+
+  const cfh = new ConfigFileHelper();
+  cfh.setCreditToAVController(creditorAddrs);
+};
+
+function validateInput(creditors: string[], privateVaults: string[], config: Config) {
```
Comment (spicysquid168) [inline deploy/exec/av-controller/config/set-param.ts:76]: I think you can create a class to map human readable input to addresses and put this function there.

File: deploy/exec/av-controller/deploy/av-controller.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+  ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+  ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+  ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+  ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+  Check all variables below before execute the deployment script
+  */
+
+  const deployer = await getDeployer();
+  console.log(">> Deploying an upgradable AutomatedVaultController contract");
+  const AutomatedVaultController = (await ethers.getContractFactory(
```
Comment (spicysquid168) [inline deploy/exec/av-controller/deploy/av-controller.ts:22]: Better use `UpgradeableContractDeployer` class

File: deploy/exec/creditor/deploy/xalpaca-creditor.ts [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+  //XALPACA bsc 0xB7d85Ab25b9D478961face285fa3D8AAecAd24a9
+  //XALPACA ftm 0x9e698f779Cec7F42663B051Ff8176A55FCb8d471
+  const XALPACA_ADDRESS = "0xB7d85Ab25b9D478961face285fa3D8AAecAd24a9";
+
+  const deployer = await getDeployer();
+
+  console.log(">> Deploying an upgradable XALPACACreditor contract");
+
+  const XALPACACreditor = (await ethers.getContractFactory("xALPACACreditor", deployer)) as XALPACACreditor__factory;
+  const xalpacaCreditor = (await upgrades.deployProxy(XALPACACreditor, [
```
Comment (spicysquid168) [inline deploy/exec/creditor/deploy/xalpaca-creditor.ts:32]: use `UpgradeableContractDeployer` class

File: deploy/exec/delta-neutral-vault-config/config/set-controller.ts [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+  */
+
+  const DELTA_VAULT_SYMBOL = ["n8x-BNBUSDT-PCS2"];
+
+  const deployer = await getDeployer();
+  const config = getConfig();
+  const CONTROLLER = config.AutomatedVaultController!.address;
+
+  // VALIDATING ALL DLTA_VAULT_SYMBOL
+  const configs = DELTA_VAULT_SYMBOL.map((inputVaultSymbol) => {
```
Comment (spicysquid168) [inline deploy/exec/delta-neutral-vault-config/config/set-controller.ts:26]: Same as previous comment. I think it is a good time to make a class to map from human readable to addresses.

Comment (spicysquid168) [reply deploy/exec/delta-neutral-vault-config/config/set-controller.ts:26]: Or even the object of the contract

**Review by alpanaca — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Review by sora-alpaca — APPROVED** [review]
